alibi_detect/saving/tests/models.py

```diff
@@ -93,18 +102,43 @@ def kernel(request, backend):
     Gaussian RBF kernel for given backend. Settings are parametrised in the test function.
     """
     kernel = request.param
-    if kernel is None:
-        pass
-    elif isinstance(kernel, dict):  # dict of kwargs
+
+    if isinstance(kernel, dict):  # dict of kwargs
+        kernel_cfg = kernel.copy()
+        sigma = kernel_cfg.pop('sigma', None)
         if backend == 'tensorflow':
-            kernel = GaussianRBF_tf(**kernel)
+            if sigma is not None and not isinstance(sigma, tf.Tensor):
+                sigma = tf.convert_to_tensor(sigma)
+            kernel = GaussianRBF_tf(sigma=sigma, **kernel_cfg)
         elif backend == 'pytorch':
-            kernel = GaussianRBF_pt(**kernel)
+            if sigma is not None and not isinstance(sigma, torch.Tensor):
+                sigma = torch.tensor(sigma)
+            kernel = GaussianRBF_pt(sigma=sigma, **kernel_cfg)
         else:
             pytest.skip('`kernel` only implemented for tensorflow and pytorch.')
     return kernel
 
 
+@fixture
+def optimizer(request, backend):
```

Comment:
This has been added so that we can test the full e2e save/load of an `optimizer` kwarg. This is only used in the `test_save_classifier` for now.